scipy/special/meson.build

```diff
@@ -25,10 +25,18 @@ _ufuncs_pxi_pxd_sources = [
   fs.copyfile('_ufuncs_extra_code_common.pxi'),
 ]
 
+if meson.get_compiler('cpp').get_id() == 'msvc'
+  sf_error_state_extra_args = ['-DSCIPY_DLL_EXPORTS']
+  sf_error_state_consumer_extra_args = ['-DSCIPY_DLL_IMPORTS']
+else
+  sf_error_state_extra_args = []
+  sf_error_state_consumer_extra_args = []
+endif
+
 sf_error_state_lib = shared_library('sf_error_state',
   ['sf_error_state.c'],
   include_directories: ['../_lib', '../_build_utils/src'],
-  cpp_args: ['-DSP_SPECFUN_ERROR'],
+  cpp_args: ['-DSP_SPECFUN_ERROR'] + sf_error_state_extra_args,
```

Comment:
`-DSP_SPECFUN_ERROR` and `DSCIPY_DLL_EXPORTS`/`DSCIPY_DLL_EXPORTS` always go together, so it'd be better to combine them in a single variable in the `if meson.get_compiler('cpp').get_id() == 'msvc' block, and remove the separate `'-DSP_SPECFUN_ERROR'` usages

Reply:
@steppi did you see this comment? Unless I'm mistaken about these flags always going together, this would be nice to address.

Reply:
Sorry, I saw it but forget I didn’t respond. I don’t think these always go together. `-DSPECUN_ERROR` goes in the `cpp_args` of all extensions which use special function error handling, regardless of platform. If it isn’t set, then  all calls to `set_error` in special become no-ops. The DLL related args should only be passed in when the compiler is `msvc ` or one of the other windows compilers . There are also some places where the DLL related args need to go into `c_args` ( for the C `sf_error_state` shared library itself of course, but also in all extensions which have C source files which use special error handling.)

Reply:
That doesn't quite mean they don't go together. Here is what I meant: https://github.com/rgommers/scipy/commit/e1206e30a20b22eb5ccb430573ef12041f95906d

Works for me locally, and does what you describe I believe. I think it should work with MSVC too. Looks a bit cleaner to me, but not a blocking comment - if you think it helps please use it, if not then let's merge this as is.

Reply:
Ah, got it. Thanks, that does look better. I’ll use it.